docs/notAlicense.txt

```diff
@@ -0,0 +1,19 @@
+notAlicense
+
+notAlab 2012-present
+
+Software and associated documentation files (the "Software") published under this license is originally released under MIT license (see below).
+
+notAlicense allows embedding of the Software into GNU GPL licensed projects. Publishing of the Software in such projects does not restrict any use of the original version of the Software under MIT license.
```

Comment:
That's in fact GPL incompatible... See:

https://en.wikipedia.org/wiki/License_compatibility#GPL_compatibility

MIT license is GPL compatible (in the context of MIT code embedded in GPL code) precisely because MIT code is then overridden.

In other words, notAlicense is actually "not a license for GPL projects", which is specifically forbidding the usage of your code inside GPL software.

How to fix that? Super-easy! Just relicense all your codes as MIT... Then, we are fulfilling everything:

1.- We are keeping the original copyright note, the very only restriction of MIT
2.- Out license states that code taken from this repo is GPL, regardless of the original license.

Reply:
This is exactly what is notAlicense saying :) The life of the code from now is under whatever global license the product is using.  Ín license I just explicitly say that original version of my code is MIT so I can freely distribute that (and only that) version of code under MIT. Any derivative work which is produced by anyone else in s44 repository will use that whatever global license with all consequences - in this case GPL vN.

Reply:
Then why not use MIT directly? It is GPL compatible: https://www.gnu.org/licenses/license-list.en.html#X11License

Your notalicense just adds confusion and more text to the repo.

Reply:
notAlicense is solving the potential uncertainty regarding the first public publishing of the code in globally GPL environment - so in fact it is reducing the confusion. 

I hope "more text to the repo" was not intended as a serious argument.

Reply:
No Pepe, is not... The origin of the software is protected by MIT license, which states that the original copyright notice shall be preserved.

Your text is making the license GPL incompatible. You are spoecifically saying that "Publishing of the Software in such projects does not restrict any use of the original version of the Software under MIT license.", and indeed if we are not modifying the software then we cannot redistribute it as GPL... You see the problem?

Reply:
1) I make original version of the software Xv1. 
2) Publish it Xv1 as notAlicense(MIT) on my repo.
3) I publish Xv1 as notAlicense(MIT) on s44 repo.
4) From that moment Xv1 is GPL positive on your repo because thats the licensing scheme of your distribution, you are allowed to distribute it under GPL from the beginning.
5) I just say that Xv1 storred on my repo is still MIT no matter what happens with X in s44 repo.

Reply:
@PepeAmpere : you have a strange vision of licensing software :-D Contact the FSF but I'm not sure it will agree with you.

btw: saying : notAlicense(MIT) is not allowed by the MIT license. To really be GPL compatible and to not break the rules set by the MIT, you need to have just this text : 
```
Copyright 2018 NotaLab, Sping1944

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```

Other are not allowed.

Reply:
Licencing software is not a game btw... it's all the more serious we are speaking about free softwares.  Renaming a licence make it proprietary and not-GPL-Compatible. The ONLY thing written where the name of the orga can appear is the first line : 
> Copyright (C) \<date\> \<project\> 

But you can't call this NotaLicense is not really allowed , in your work or in this repo. It's forbidden.

example of an MIT license from a company : https://github.com/roadiz/roadiz/blob/master/LICENSE.md

Reply:
No notalicense in spring1944 please. @FLOZi @kanatohodets

Reply:
Pepe, again, you are not saying that:

> ... does not restrict any use of the **original version of the Software** ...

And again, the MIT license already make the cover you are trying to get.

Of course, I concern here about the integration in S44. Said that, as a friend I would warn you that your license is even MIT incompatible. Personally I guess your license changes don't worth eventual legal issues...

We can discuss further if you want, but I'll complain as many as required to avoid getting integrated software under notAlicense terms. The last thing I want in my life is a lawyer threatening me

Reply:
Wow, busy PR! Yes, please no custom licenses. [Spring's stance](https://springrts.com/phpbb/viewtopic.php?f=1&t=17847) is pretty clear: Spring-related Lua code is considered linking against the engine, and so those works must be GPL, full-stop. I think the simplest thing is for _any_ Spring Lua code to be GPL licensed from the outset; any more concrete recommendation than that starts to get into long hours of reading license compatibility write-ups and playing lawyer.

From a practical standpoint using custom variations of well-known licenses is likely to get the work excluded from free software repositories: Debian, for example, just doesn't have time or legal risk tolerance to closely read your license and confirm its legal integrity. There's a can of worms regarding these repositories and the CC-BY-NC license that the art is under, but there's no good reason to make that situation worse.

In terms of what the code was originally published as -- this doesn't matter if you're the sole copyright holder. You can just contribute it to this project under the terms of an entirely different license.

Reply:
@PepeAmpere has already removed the notAlicense reference in the internal code, but the submodules are still licensed under notAlicense GPL incompatible terms. Those codes are in fact downloaded and deployed within the game each time you log in the host, so a license collision remains.